src/SFA.DAS.LevyTransferMatching.Api/Controllers/PledgeController.cs

```diff
@@ -20,13 +21,37 @@ public PledgeController(IMediator mediator)
 
         [HttpGet]
         [Route("pledges")]
+        [ProducesResponseType((int)HttpStatusCode.OK)]
         public async Task<IActionResult> GetPledges()
         {
             var result = await _mediator.Send(new GetPledgesQuery());
 
             return new OkObjectResult(result.Select(x => (PledgeDto)x));
         }
-		
+
+        [HttpGet]
+        [Route("pledges/{pledgeId}")]
+        [ProducesResponseType((int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        public async Task<IActionResult> GetPledge(int pledgeId)
+        {
+            var result = await _mediator.Send(new GetPledgesQuery()
```

Comment:
As with the api layer, this should be a separate query with _nothing_ shared

Reply:
This has now been changed, as requested.